lib/scope.js

```diff
@@ -264,23 +271,21 @@ module.exports = {
 
     if ( scope.page ) {
       let { id } = await scope.examinePageID( scope, 'none to match' );
-      name += `${ id }-`;
+      let short_id = `${ id }`.substring(0, 20);
+      name += `${ short_id }-`;
     }
 
     name += scope.base_filename;
-    name += `-${ files.readable_date() }`;
+    name += `-${ Date.now() }`;
 
     let safe_name = safe_filename( name, { replacement: `_` });
-    // Allow room for extensions, etc.
-    let safer_name = safe_name.substring(0, (255 - 10));
-    return safer_name;
+    return safe_name;
   },  // Ends scope.getSafeScenarioFilename()
 
   getJSONFilename: async function( scope ) {
     // Get a unique path for a json file
     let { id } = await scope.examinePageID( scope, 'none to match' );
-    let date = files.readable_date();
-    let json_path = path.join( scope.paths.scenario, `json_for-${ id }-${ date }.json` );
+    let json_path = path.join( scope.paths.scenario, `json_for-${ id }-${ Date.now() }.json` );
```

Comment:
Need to shorten the id in the json path name. Gorammit. I think we might update this for v5, but not v4. New issue.